File: src/processes/textProcessing.js
const didYouMean = require("google-did-you-mean");

function removePunctuation(text) {
  return text.replace(/[.,;:?!'"]/g, "").toLowerCase();
}

async function correctSpelling(word) {
  const query = await didYouMean(word);
  return query.suggestion != null ? query.suggestion : word;
}

async function correctSentence(sentence) {
  const words = sentence.split(/\s+/);
  const correctedWords = await Promise.all(
    words.map((word) => correctSpelling(word)),
  );
  return correctedWords.join(" ");
}

async function main(sentence) {
  text = removePunctuation(sentence);
  text = await correctSentence(text);
}

module.exports = {
  main,
};

File: src/processes/textProcessing.py
import re

from spellchecker import SpellChecker


def remove_punctuation(text):
    return re.sub(r'.,;:?!\'"', "", text).lower().strip()


def correct_spelling(word):
    spell = SpellChecker()
    return spell.correction(word)


def remove_punctuation_sync(text):
    return remove_punctuation(text)


def correct_sentence(sentence):
    words = remove_punctuation_sync(sentence)
    corrected_words = [correct_spelling(word) for word in words.split()]
    return " ".join(corrected_words)
